compiler/rustc_resolve/src/diagnostics.rs

```diff
@@ -330,20 +335,20 @@ impl<'a, 'tcx> Resolver<'a, 'tcx> {
 
         match import {
             Some((import, span, true)) if should_remove_import && import.is_nested() => {
-                self.add_suggestion_for_duplicate_nested_use(&mut err, import, span)
+                self.add_suggestion_for_duplicate_nested_use(&mut err, import, span);
             }
             Some((import, _, true)) if should_remove_import && !import.is_glob() => {
                 // Simple case - remove the entire import. Due to the above match arm, this can
                 // only be a single use so just remove it entirely.
-                err.tool_only_span_suggestion(
-                    import.use_span_with_attributes,
-                    "remove unnecessary import",
-                    "",
-                    Applicability::MaybeIncorrect,
+                err.subdiagnostic(
+                    self.tcx.dcx(),
+                    errors::ToolOnlyRemoveUnnecessaryImport {
+                        span: import.use_span_with_attributes,
+                    },
```

Comment:
Hmm, is this conversion correct? docs for `tool_only_span_suggestion` says

> Adds a suggestion to the JSON output that will not be shown in the CLI.

Which I assume has different behavior from simply `err.subdiagnostic()` here?

Reply:
The impl for `tool_only_span_suggestion` is

```rs
#[rustc_lint_diagnostics]
pub fn tool_only_span_suggestion(
    &mut self,
    sp: Span,
    msg: impl Into<SubdiagMessage>,
    suggestion: impl ToString,
    applicability: Applicability,
) -> &mut Self {
    self.span_suggestion_with_style(
        sp,
        msg,
        suggestion,
        applicability,
        SuggestionStyle::CompletelyHidden,
    );
    self
}
```

Note the `SuggestionStyle::CompletelyHidden`. I wonder if you'll need a `err.tool_only_subdiagnostic()`?

Reply:
`ToolOnlyRemoveUnnecessaryImport` has the following attribute :
```rust
#[suggestion(
    resolve_remove_unnecessary_import,
    code = "",
    applicability = "maybe-incorrect",
    style = "tool-only"
)]```
which should not show the suggestion in the CLI.

Reply:
Ah my bad, I completely overlooked that singular attribute (I was constantly jumping back and forth between the derives and the diagnostic construction sites lol).